refactor(compiler-cli):  add warning for matching viewport `@defer` t…

SkyZeroZx opened the pull request:
…rigger options

Ensures that viewport `@defer` trigger prefetch warnings occur when the viewport options are identical, including cases with empty or no parameters.
 
 Currently, when using cases like don’t show warning, this PR adds proper support and fixes the missing viewport warning.

With viewport options
 ```
     @defer (
      on viewport({   rootMargin: '100px' , threshold: 0.1});
      prefetch on viewport({    threshold: 0.1  rootMargin: '100px'})
    ) {
      <p>Defer Content</p>
    }  @placeholder {
      <p>Placeholder</p>
    }
 ```
 
Using default viewport
  ```
     @defer (
      on viewport;
      prefetch on viewport // also viewport({})
    ) {
      <p>Defer Content</p>
    }  @placeholder {
      <p>Placeholder</p>
    }
 ```

JeanMeche commented:
/gemini review

SkyZeroZx commented on packages/compiler-cli/src/ngtsc/typecheck/extended/checks/defer_trigger_misconfiguration/index.ts:
```diff
@@ -16,12 +16,39 @@ import {
+  }
+
+  return true;
+}
```
I'm not very sure about modifying or changing it; Gemini's proposal is slower than the proposed implementation (although more readable). Which one should we choose?
Or does efficiency really not matter much in this case?

JeanMeche replied:
Perf is somehow important when work around the compiler. But here the size of the maps we check is fairly small, I don't think there is a significant difference between both implementation. Your call.

SkyZeroZx replied:
I'd prefer to stick with the one I proposed, considering it's faster and doesn't significantly affect readability since it's a helper. I've updated the tests with refactoring for better readability.

SkyZeroZx committed:
refactor(compiler-cli): add warning for matching viewport @defer trigger options

Ensures that viewport `@defer` trigger prefetch warnings occur when the viewport options are identical, including cases with empty or no parameters.
